src/nupic/algorithms/TemporalMemory.cpp

```diff
@@ -460,7 +456,10 @@ static void punishPredictedColumn(
 }
 
 void TemporalMemory::activateCells(const SDR &activeColumns, bool learn) {
-    NTA_CHECK( activeColumns.dimensions == columnDimensions_ );
+    NTA_CHECK( activeColumns.dimensions.size() == columnDimensions_.size() );
+    for(size_t i=0; i< columnDimensions_.size(); i++) {
+      NTA_CHECK(static_cast<size_t>(activeColumns.dimensions[i]) == static_cast<size_t>(columnDimensions_[i])) << "Dimensions must be the same.";
+    }
```

Comment:
Can we keep the shorter version?  Also the comments.
`NTA_CHECK( activeColumns.dimensions == columnDimensions_ ) << "Dimensions must be the same.";`

Reply:
> Can we keep the shorter version? 

unfortunately no, typedefs in Connections allow a change (CellIdx, and I'd prefer to keep CellIdx even the type for columnDimensions, as it directly tells the user the range. ) 
- while SDR::dimensions is UInt

- so I was getting unsupported compare `vector<uint32> == vector<uint16>`;
  - if we do element-by-element, NTA_CHECK is able to do the comparison

Reply:
Ok, that makes sense.  Maybe in a future PR:  SDR.dimsnsions could use CellIdx too?